mealpy.py

```diff
@@ -121,14 +128,72 @@ def cancel_current_meal(self):
         raise NotImplementedError()
 
 
-SCHEDULER = BlockingScheduler()
+def get_mealpal_credentials():
+    email = CONFIG['email_address']
+    if CONFIG['use_keyring']:
+        password = (
+            keyring.get_password(KEYRING_SERVICENAME, email)
+            or getpass.getpass('Credential not yet stored in keychain, '
+                               'please enter password: ')
+        )
+        keyring.set_password(KEYRING_SERVICENAME, email, password)
+    else:
+        password = getpass.getpass('Enter password: ')
+    return MealPal(email, password)
+
+
+@click.group()
+@click.pass_context
+def login_group(ctx):
+    mealpal = get_mealpal_credentials()
+    ctx.obj['mealpal'] = mealpal
 
 
-@SCHEDULER.scheduled_job('cron', hour=16, minute=59, second=58)
-def execute_reserve_meal():
+@click.group()
+def non_login_group():
+    pass
 
-    mealpal = MealPal(EMAIL, PASSWORD)
 
+@non_login_group.command('save_pass', short_help='Save a password into the keyring.')
+def save_pass():
+    keyring.set_password(KEYRING_SERVICENAME, CONFIG['email_address'],
+                         getpass.getpass('Enter password: '))
+    print('Password successfully saved to keyring.')
+
+
+@login_group.command('get_restaurants',
+                   short_help='Display all available restaurants for the '
+                              'current window.')
+@click.pass_context
+def get_restaurants(ctx):
+    mealpal = ctx.obj['mealpal']
+    while True:
+        status_code = mealpal.login()
+        if status_code == 200:
+            print('Logged In!')
+            break
+        else:
+            print('Login Failed! Retrying...')
+
+    # Fetch all restaurants and allow interactive search on restaurants
+    pass
+    # import pdb; pdb.set_trace()
```

Comment:
Debug statements

Reply:
I think I'll remove `get_restaurants` for now since it's not doing anything much anyway.